c2cgeoportal/scripts/c2ctool.py

```diff
@@ -344,32 +346,86 @@ def step1(self):
         check_call(["make", "-f", self.options.file, self.options.clean])
 
         diff_file = open("changelog.diff", "w")
-        check_call(["git", "diff", "CONST_CHANGELOG.txt"], stdout=diff_file)
+        check_call(["git", "diff", "--", "CONST_CHANGELOG.txt"], stdout=diff_file)
         diff_file.close()
 
         check_call(["make", "-f", self.options.file, "update-node-modules"])
         check_call(["make", "-f", self.options.file, ".build/requirements.timestamp"])
 
-        print("")
-        print(self.color_bar)
-        print("\n".join(notes))
-        print(
-            "Apply the manual migration steps based on what is in the CONST_CHANGELOG.txt file"
-            " (listed in the `changelog.diff` file)."
-        )
-        self.print_step(2)
+        if os.path.getsize("changelog.diff") == 0:
+            self.step2()
+        else:
+            print("")
+            print(self.color_bar)
+            print(
+                "Apply the manual migration steps based on what is in the CONST_CHANGELOG.txt file"
+                " (listed in the `changelog.diff` file)."
+            )
+            self.print_step(2)
 
     def step2(self):
+        if os.path.isfile("changelog.diff"):
+            os.unlink("changelog.diff")
+
+        diff_file = open("ngeo.diff", "w")
```

Comment:
Just a comment, you don't have to act on it...

I prefere that construct (safer and more compact):
```python
with open("ngeo.diff", "w") as diff_file:
    ...
```